app/src/test/java/tech/ula/model/repositories/AssetRepositoryTest.kt

```diff
@@ -0,0 +1,140 @@
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.times
+import org.mockito.junit.MockitoJUnitRunner
+import tech.ula.model.entities.Asset
+import tech.ula.utils.AssetListPreferenceUtility
+import tech.ula.utils.ConnectionUtility
+import tech.ula.utils.TimestampPreferenceUtility
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.io.InputStream
+
+@RunWith(MockitoJUnitRunner::class)
+class AssetRepositoryTest {
+
+    @get:Rule
+    val tempFolder = TemporaryFolder()
+
+    @Mock
+    lateinit var timestampPreferenceUtility: TimestampPreferenceUtility
+
+    @Mock
+    lateinit var assetListPreferenceUtility: AssetListPreferenceUtility
+
+    @Mock
+    lateinit var connectionUtility: ConnectionUtility
+
+    lateinit var applicationFilesDirPath: String
+
+    private val archType = "arch"
+    private val distType = "dist"
+    val allAssetListTypes = listOf(
+            "support" to "all",
+            "support" to archType,
+            distType to "all",
+            distType to archType
+    )
+
+    lateinit var assetRepository: AssetRepository
+
+    @Before
+    fun setup() {
+        applicationFilesDirPath = tempFolder.root.path
+        assetRepository = AssetRepository(archType, distType, applicationFilesDirPath,
+                timestampPreferenceUtility, assetListPreferenceUtility, connectionUtility)
+    }
+
+    @Test
+    fun allTypesOfCachedAssetListsAreRetrieved() {
+        assetRepository.getCachedAssetLists()
+        verify(assetListPreferenceUtility).getAssetLists(allAssetListTypes)
+    }
+
+    @Test
+    fun fetchingDistributionAssetListsRemovesRootfsFiles() {
+        val distTypeAssetLists = allAssetListTypes.filter {
+            (assetType, _) ->
+            assetType == distType
+        }
+        val asset1 = Asset("name", distType, archType, 0)
+        val asset2 = Asset("rootfs.tar.gz", distType, archType, 0)
+        val assetListWithRootfsFile = listOf(listOf(asset1, asset2))
+        `when`(assetListPreferenceUtility.getAssetLists(distTypeAssetLists)).thenReturn(assetListWithRootfsFile)
+
+        val returnedAssetList = assetRepository.getDistributionAssetsList(distType)
+
+        assertTrue(returnedAssetList.size == 1)
+        assertFalse(returnedAssetList.any { it.name == "rootfs.tar.gz" })
+    }
+
+    @Test
+    fun formatsUrlBasedOnHttpsAccessibility() {
+        val inputStream = ByteArrayInputStream("asset 0".toByteArray()) as InputStream
+        `when`(connectionUtility.getUrlInputStream(anyString())).thenReturn(inputStream)
+
+        val allUrlsWithoutProtocols = allAssetListTypes.map { (dist, arch) ->
+            "://github.com/CypherpunkArmory/UserLAnd-Assets-" +
+                    "$dist/raw/master/assets/$arch/assets.txt"
+        }
+
+        argumentCaptor<String>().apply {
+            assetRepository.retrieveAllRemoteAssetLists(httpsIsAccessible = true)
+            verify(connectionUtility, times(4)).getUrlInputStream(capture())
```

Comment:
the `4` here seems like a number out of no where.  Is it related to the number of elements in `allAssetListTypes`?

Can we put it into a variable for easier readability?